packages/lesswrong/components/ea-forum/digest/EditDigestHeader.tsx

```diff
@@ -0,0 +1,107 @@
+import React, { useState } from 'react';
+import { Components, registerComponent } from '../../../lib/vulcan-lib';
+import { getDigestName } from '../../../lib/collections/digests/helpers';
+import moment from 'moment';
+import { useUpdate } from '../../../lib/crud/withUpdate';
+import classNames from 'classnames';
+
+const styles = (theme: ThemeType): JssStyles => ({
+  root: {
+    display: "flex",
+    gap: "4px",
+  },
+  date: {
+    cursor: "pointer",
+    "&:hover": {
+      color: theme.palette.grey[800],
+    },
+  },
+  datePicker: {
+    transform: 'translateY(-12px)',
+    zIndex: 2
+  }
+});
+
+export const EditDigestHeader = ({digest, refetchPosts, classes}: {
+  digest: DigestsMinimumInfo,
+  refetchPosts: () => void,
+  classes: ClassesType,
+}) => {
+  // clicking on the start or end date lets you edit it
+  const [isEditingStartDate, setIsEditingStartDate] = useState(false);
+  const [isEditingEndDate, setIsEditingEndDate] = useState(false);
+  const {mutate: updateDigest} = useUpdate({
+    collectionName: "Digests",
+    fragmentName: "DigestsMinimumInfo",
+  });
+
+  const startFormatted = moment(digest.startDate).format('MMM D')
+  const endFormatted = digest.endDate ? moment(digest.endDate).format('MMM D') : 'now'
+
+  const onChangeDate = (field: string, date?: Date) => {
+    if (date) {
+      void updateDigest({
+        selector: {_id: digest._id},
+        data: {
+          [field]: date,
+        },
+      });
+    }
+  }
+
+  const {SectionTitle, DatePicker} = Components;
+
+  const startNode = isEditingStartDate
+    ? (
+      <div className={classes.datePicker}>
+        <DatePicker
+          label="Start date"
+          value={new Date(digest.startDate)}
+          onChange={onChangeDate.bind(null, "startDate")}
+          onClose={refetchPosts}
```

Comment:
Very nitpicky, but I think you should set `isEditingStartDate` to false on close.

Reply:
Seems reasonable, I did that instead of the refetch

Reply:
Why instead?

Reply:
Oh turns out I didn't need to call refetch, already does so when updating the digest dates